compiler/include/expr.h

```diff
@@ -381,13 +381,11 @@ static inline bool isAliveQuick(Symbol* symbol) {
 }
 
 static inline bool isAlive(Symbol* symbol) {
-  if (symbol->hasFlag(FLAG_GLOBAL_TYPE_SYMBOL)) return true;
-  if (! symbol->defPoint) return false;
-  return isAliveQuick(symbol);
+  return symbol->defPoint && isAlive(symbol->defPoint);
```

Comment:
The case for FLAG_GLOBAL_TYPE_SYMBOL was added in cd44f189 without an explanation in the commit message. 
I will extend the PR message with an explanation of why I think it is OK to remove it.

isAliveQuick contains one extra branch compared to isAlive.